### PR TITLE
add: login session extension feature with UI button

### PR DESCRIPTION
**Pull request by lizable**
Related: [backend.ai#2456](https://github.com/lablup/backend.ai/pull/2456)
This PR refactors TimeContainer component to be used as counting down valid login session time received from server.

1. See until the time passes on right side of header.
2. Click `Extend` button on the right top of header.
3. Wait until 5 or 3 seconds left and Click the `Extend` button again.
4. See session login time extension works.

### Screenshot(s):
Screenshot 2024-07-26 at 8 13 50 PM

**Checklist:** (if applicable)

- [x] Mention to the original issue
- [ ] Documentation
- [x] Minimum required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [x] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by lizable** [comment]
* **#2551** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2551?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lizable and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/TimeContainer.tsx [hunk 1]
```diff
@@ -1,22 +1,64 @@
...
+    onSuccess: (data) => {
+      setExpires(data.expires);
+    },
+  });
 
   useEffect(() => {
-    const timer = setInterval(() => setDate(new Date()), 1000);
+    refetch();
+    const timer = setInterval(() => {
+      setExpires((prevExpires) => prevExpires - 1); // 1초씩 감소
```
Comment (agatha197) [inline react/src/components/TimeContainer.tsx]:
```suggestion
      setExpires((prevExpires) => prevExpires - 1); // Decrease by 1 second each time.
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
Please check the [test](https://github.com/lablup/backend.ai-webui/actions/runs/10110244267/job/27959772213?pr=2551)

**Review by agatha197 — CHANGES_REQUESTED** [review]
When set to 15 seconds, extending it results in 14 seconds (which is correct), but refreshing extends it to 13 seconds. Both should be extended to 14 seconds.

File: react/src/components/TimeContainer.tsx [hunk 2]
```diff
@@ -1,22 +1,59 @@
...
-    </>
+    <Flex direction="row" gap="xxs">
+      <ClockCircleOutlined />
+      <BAIIntervalText
+        callback={() => {
+          const diff = dayjs(data?.expires).diff(dayjs(), 'seconds');
+          const duration = dayjs.duration(Math.max(0, diff), 'seconds');
+          return duration.format(format);
+        }}
+        delay={1000}
```
Comment (yomybaby) [inline react/src/components/TimeContainer.tsx:48]: To display more accurate time:
```suggestion
        delay={100}
```

**Review by yomybaby — CHANGES_REQUESTED** [review]

File: react/src/components/TimeContainer.tsx [hunk 3]
```diff
@@ -1,22 +1,59 @@
...
+    queryFn: () => {
+      return baiRequestWithPromise({
+        method: 'POST',
+        url: `/server/extend-login-session`,
+      });
+    },
+    staleTime: 0,
+    cacheTime: 0,
+    suspense: true,
   });
```
Comment (yomybaby) [inline react/src/components/TimeContainer.tsx:37]:
```suggestion
  const { data } = useSuspenseQuery<{
    expires: string;
  }>({
    queryKey: ['TimeContainerExpires'],
    queryFn: () => {
      return baiRequestWithPromise({
        method: 'POST',
        url: `/server/extend-login-session`,
      });
    },
    staleTime: 0,
    cacheTime: 0,
    fetchKey
  });
```

File: react/src/components/TimeContainer.tsx [hunk 4]
```diff
@@ -1,22 +1,59 @@
...
+import { useTranslation } from 'react-i18next';
 
-interface TimeContainerProps {}
+interface TimeContainerProps {
+  format?: string;
+}
 
-const TimeContainer: React.FC<TimeContainerProps> = () => {
-  const [date, setDate] = useState(new Date());
+const TimeContainer: React.FC<TimeContainerProps> = ({
```
Comment (yomybaby) [inline react/src/components/TimeContainer.tsx:16]: It's better to use a more explicit name. `TimeContainer` => `LoginSessionExtendButton`

**Review by yomybaby — CHANGES_REQUESTED** [review]

**Review by yomybaby — APPROVED** [review]
LGTM

File: config.toml.sample [hunk 5]
```diff
@@ -30,8 +30,9 @@ isDirectorySizeVisible = true           # If false, directory size in folder exp
 maxCountForPreopenPorts = 10            # The maximum allowed number of preopen ports. If you set this option to 0, the feature of preopen ports is disabled.
 allowCustomResourceAllocation = true    # If true, display the custom allocation on the session launcher.
 eduAppNamePrefix = ""                   # The prefix of edu applauncher's app name. If the app name starts with this prefix, split it by '-' and use the tail as the name of image.
-enableModelStore = false    		        # Enable model store feature. (From Backend.AI 24.03)
+enableModelStore = false                # Enable model store feature. (From Backend.AI 24.03)
 enableLLMPlayground = false             # Enable LLM Playground feature. (From Backend.AI 24.03)
+enableExtendLoginSession = false        # If true, enables login session extension UI. (From Backend.AI 24.03.7)
```
Comment (agatha197) [inline config.toml.sample]:
```suggestion
enableExtendLoginSession = false        # If true, enables login session extension UI. (From Backend.AI 24.09)
```

**Review by agatha197 — CHANGES_REQUESTED** [review]

**Review by agatha197 — APPROVED** [review]
LGTM!

**Pull request merged**